src/nvim/screen.c

```diff
@@ -1332,6 +1333,9 @@ static void win_update(win_T *wp, Providers *providers)
   srow = 0;
   lnum = wp->w_topline;  // first line shown in window
 
+  win_extmarks_arr.size = 0;
+  win_extmarks_arr.capacity = 0;
+  win_extmarks_arr.items = NULL;
```

Comment:
`kv_size(win_extmarks_arr) = 0`. don't reset the others, we want to reuse the allocated buffer.

Reply:
crashes if I do so. maybe freed after the message is sent?

Reply:
not a problem if it's not directly passed to ui event call.

Reply:
done.